plugins/elasticsearch/komand_elasticsearch/util/api.py

```diff
@@ -0,0 +1,73 @@
+import json
+import requests
+from insightconnect_plugin_runtime.exceptions import PluginException
+from requests import Response
+from requests.auth import HTTPBasicAuth
+
+
+class ElasticSearchAPI:
+    def __init__(self, url: str, logger: object, username: str = None, password: str = None):
+        self.url = url.rstrip("/")
```

Comment:
If a user provides a domain (my-es-server.com) or IP (1.2.3.4) instead of a URL (https://...: which happens sometimes), it would be good to attempt to connect to https:// and then to http:// if that doesn't work.